PicoNetworks/wordpress-plugin
# 52 plugin installation enhancements

Whoops, I made this PR with picosystem on accident -scott

### Description
Increases the visibility of available updates and streamlines updating of our plugin.
* Added an update checker lib, code is native in our repo (not using composer or anything "fancy")
* Updated the health_check api call to return with the current wordpress version
* Handled check in widget.php to avoid a code path where no HTTP_USER_AGENT is present (not too sure about why this was necessary but we can chat if any concern)

:white_check_mark: Fixes #52 
:link: Test with 
* Pub App Change https://github.com/PicoNetworks/publisher/pull/321
* API Change https://github.com/PicoNetworks/API/pull/376

### This change requires changes to environment variables
In your local `wp-config.php`, add the following line. I have already updated staging to be staging. By default, all production plugins will use the master branch. This env var enables a switch for the update checker to check against a specific git branch instead of the latest GitHub release.
```
/** pico config */
// ...
define('PP_BRANCH_NAME', '52-plugin-installation-enhancements');
// ...
```


### Instructions for review

#### Test Auto Updater
1. Ensure you have added the `php-config.php` variable described above
2. Go to https://github.com/PicoNetworks/wordpress-plugin/tree/52-plugin-installation-enhancements and download a zip of the branch
3. **Remove any symlinks** you may have in your local env, they will break the auto updater. Go to your local Wordpress admin console, delete your current plugin, and upload the zip of the plugin under test.
4. In your local Wordpress admin console, use the Plugin "Editor" to change the Version on lines 6 and 11 to "0.1.7" (this is one version behind this release of 0.1.8)
5. Go back to the wordpress plugin dashboard. If you do not see the wordpress plugin declaring an update is needed, click on the "check for updates" button.
6. You should see the Pico plugin asking to be updated. Click on the "view update details" button.
7. You should see some basic boilerplate info on changelog, description, etc. Waiting on @nychen to decided what info to include here via the `readme.txt` file documented at https://developer.wordpress.org/plugins/wordpress-org/how-your-readme-txt-works/

#### Test Staging
1. After we merge into staging, we need to follow a similar process to install the new plugin and test it. Moving forward, we should be able to utilize one-click updates for staging and prod env, just like publishers.
2. I've already added WP config var to staging so the plugin will download from the staging branch.

### Test WP Healthcheck response
1. Postman > Wordpress > Wordpress Health Check
2. Should see a response of {connected: true, pico_version: 0.1.8}

To recap, wp config variables will tell local and staging env to download from whatever branch is in the config variable `PP_BRANCH_NAME`. By default, production (master) will use the latest GitHub Release.

:pray: to be blessed and merged in by @hoppy-kamper
